src/Notifications/NotificationsHub.cs

```diff
@@ -35,14 +49,39 @@ public override async Task OnDisconnectedAsync(Exception exception)
     {
         var currentContext = new CurrentContext(null, null);
         await currentContext.BuildAsync(Context.User, _globalSettings);
+
+        var clientType = DeviceTypes.ToClientType(currentContext.DeviceType);
+        if (clientType != ClientType.All && currentContext.UserId.HasValue)
+        {
+            await Groups.RemoveFromGroupAsync(Context.ConnectionId,
+                GetUserGroup(currentContext.UserId.Value, clientType));
+        }
+
         if (currentContext.Organizations != null)
         {
             foreach (var org in currentContext.Organizations)
             {
-                await Groups.RemoveFromGroupAsync(Context.ConnectionId, $"Organization_{org.Id}");
+                await Groups.RemoveFromGroupAsync(Context.ConnectionId, GetOrganizationGroup(org.Id));
+                if (clientType != ClientType.All)
+                {
+                    await Groups.RemoveFromGroupAsync(Context.ConnectionId, GetOrganizationGroup(org.Id, clientType));
+                }
             }
         }
+
         _connectionCounter.Decrement();
         await base.OnDisconnectedAsync(exception);
     }
+
+    public static string GetUserGroup(Guid userId, ClientType clientType)
+    {
+        return $"UserClientType_{userId}_{clientType}";
+    }
+
+    public static string GetOrganizationGroup(Guid organizationId, ClientType? clientType = null)
+    {
+        return clientType is not ClientType.All
+            ? $"Organization_{organizationId}"
+            : $"OrganizationClientType_{organizationId}_{clientType}";
+    }
```

Comment:
```suggestion
    public static string GetOrganizationGroup(Guid organizationId, ClientType? clientType = null)
    {
       return clientType is ClientType.All or null
           ? $"OrganizationClientType_{organizationId}"
           : $"OrganizationClientType_{organizationId}_{clientType}";
    }
```

Wouldn't we want to include the client type if it wasn't `ClientType.All`? 

I might be confused here.

Reply:
You are right, i've got it the other way around 🤦 Will fix it in next commit.
In the scenario when `clientType is ClientType.All or null`, the Notification needs to be sent to entire Organization, regardless of client type, so the `Organization_` prefix better describes the scenario. It's also backwards compatible.

Reply:
Fixed in [491789b](https://github.com/bitwarden/server/pull/4923/commits/491789bd5a6454faed8e47f4c948ffbc05144c5b)